bluesky/plans.py

```diff
@@ -16,13 +18,41 @@
 from . import plan_patterns
 
 from . import utils
-from .utils import Msg, get_hinted_fields
+from .utils import (
+    Msg,
+    get_hinted_fields,
+    ScalarOrIterable,
+    CustomPlanMetadata,
+    MsgGenerator,
+)
 
 from . import preprocessors as bpp
 from . import plan_stubs as bps
-
-
-def count(detectors, num=1, delay=None, *, per_shot=None, md=None):
+from .protocols import Flyable, Movable, Readable
+
+#: Plan function that can be used for each shot in a detector acquisition involving no actuation
+PerShot = Callable[[Iterable[Readable], Optional[bps.TakeReading]], MsgGenerator]
+
+#: Plan function that can be used for each step in a scan
+PerStep = Callable[
+    [
+        Iterable[Readable],
+        Mapping[Movable, Any],
+        Mapping[Movable, Any],
+        Optional[bps.TakeReading],
+    ],
+    MsgGenerator,
+]
+
+
+def count(
+    detectors: List[Readable],
```

Comment:
This should probably be `Union[List[Readable], Tuple[Readable]]` for generality.

Reply:
Good point, I think `Sequence` works as a supertype of both. Would `Sequence[Readable]` work with Queue Server validation? It's much more concise.

Reply:
I made some changes to the Queue Server. Using `Iterable` is good now.

Reply:
It doesn't support `Sequence` then?

Reply:
It does.

Reply:
I'm generally in favor of "Accept lazy iterables (generators) everywhere!" but I think that it's pretty reasonable to expect a `Sequence` here.

Reply:
I think make it `Sequence` here to reflect what it will accept and make a separate issue for making all plans accept lazy iterables. https://github.com/bluesky/bluesky/issues/1613